src/backend/distributed/metadata/metadata_cache.c

```
@@ -4353,6 +4353,7 @@ GetIntervalTypeInfo(char partitionMethod, Var *partitionColumn,
 		case DISTRIBUTE_BY_APPEND:
 		case DISTRIBUTE_BY_RANGE:
 		{
+			Assert(partitionColumn != NULL);
```

should we throw an error instead ?

---

Yeah, I think that makes sense.